processor/servicegraphprocessor/processor.go

```diff
@@ -288,7 +299,25 @@ func (p *serviceGraphProcessor) onExpire(e *store.Edge) {
 		zap.String("connection_type", string(e.ConnectionType)),
 		zap.Stringer("trace_id", e.TraceID),
 	)
-	stats.Record(context.Background(), statExpiredEdges.M(1))
+
+	if p.config.VirtualNodeFeatureEnabled {
```

Comment:
This can be cleaned up a bit. The log above can be confusing if we end up recording the metric, maybe move it down to when the edge is expired and not aggregated.